### PR TITLE
Use sqlite3.SQLiteDriver directly, fix issue #73

### PR DESCRIPTION
**Pull request by zmedico**

### CODE REVIEW

File: db/db.go [hunk 1]
```diff
@@ -80,62 +69,45 @@ func Open(dbPath string) (*DB, error) {
 	openMu.Lock()
 	defer openMu.Unlock()
 
-	dbc, err := sql.Open("sqlite3_with_hook", dbPath)
+	os.MkdirAll(filepath.Dir(dbPath), 0755)
```
Comment (otoolep) [inline db/db.go]: The error should be checked here.

File: db/db.go [hunk 2]
```diff
@@ -80,62 +69,45 @@ func Open(dbPath string) (*DB, error) {
...
+	d := sqlite3.SQLiteDriver{}
+	dbc, err := d.Open(dbPath)
+
```
Comment (otoolep) [inline db/db.go]: Nit: please remove this blank line.

File: db/db.go [hunk 3]
```diff
@@ -17,21 +18,10 @@ import (
 // to the driver's internal conn object. The mutex ensures only 1 Open() call
 // can run concurrently, and when it has returned, sqlite3conn will be set to
 // the connection. This connection can be used for backups.
-var sqlite3conn *sqlite3.SQLiteConn
 var openMu sync.Mutex
```
Comment (otoolep) [inline db/db.go]: This mutex and all the comment above can be removed now, right?

File: db/db.go (hunk 1 above)
Comment (otoolep) [inline db/db.go]: Also, I don't think this is necessary. The directory tree will be created, right?

File: db/db.go [hunk 4]
```diff
@@ -80,62 +69,45 @@ func Open(dbPath string) (*DB, error) {
...
 // Execute executes queries that modify the database.
 func (db *DB) Execute(queries []string, tx, xTime bool) ([]*Result, error) {
 	type Execer interface {
-		Exec(query string, args ...interface{}) (sql.Result, error)
+		Exec(query string, args []driver.Value) (driver.Result, error)
 	}
 
 	var allResults []*Result
-	err := func() error {
+	err := func() (err error) {
```
Comment (otoolep) [inline db/db.go]: Why did you decide to name the error?

**Comment by otoolep** [comment]
Thanks for doing this @zmedico -- it's much better. I do have some feedback before I can merge however.

**Comment by otoolep** [comment]
I also started a CHANGELOG. Can you update it?

https://github.com/otoolep/rqlite/blob/master/CHANGELOG.md
 
Add the following line to the 2.0 section:

```
- [PR 74](https://github.com/otoolep/rqlite/pull/74): Use SQLite3 connection directory
```

File: db/db.go (hunk 1 above)
Comment (zmedico) [inline db/db.go]: Apparently sql.Open creates the directory. I just tested again, and this is the error if we don't create the directory:
```2016/03/14 21:05:39 failed to open store: unable to open database file```

File: db/db.go (hunk 4 above)
Comment (zmedico) [inline db/db.go]: If I don't name the error, then I get this:
```
src/github.com/otoolep/rqlite/db/db.go:126: undefined: err
src/github.com/otoolep/rqlite/db/db.go:127: undefined: err
src/github.com/otoolep/rqlite/db/db.go:128: undefined: err
```
If use `:=` to try and fix the above problem, I get this:
```
src/github.com/otoolep/rqlite/db/db.go:126: t declared and not used
```
I'm open to suggestions if there is a better way to achieve the intended result.

File: db/db.go (hunk 1 above)
Comment (otoolep) [inline db/db.go]: OK, I see what you're saying -- my mistake.

The directory creation should be performed up a level, which I've done here: https://github.com/otoolep/rqlite/commit/e7e212411a4831fa3b24aa3a02b102ad162ac431

That way you don't need to worry about it now.

Comment (zmedico) [inline db/db.go]: Updated to check for MkdirAll error.

Comment (otoolep) [inline db/db.go]: Just to explain my reasoning. The directory is also needed by the Store, but the Store is more fundamental to the system than the SQLite file, hence have the Store create it.

File: db/db.go (hunk 2 above)
Comment (zmedico) [inline db/db.go]: Done.

File: db/db.go [hunk 5]
```diff
@@ -77,65 +57,48 @@ type Rows struct {
...
 		var execer Execer
 		var rollback bool
+		var t driver.Tx
```
Comment (otoolep) [inline db/db.go]: OK, for the sake of consistency, how about just doing `var err error` here?

File: db/db.go (hunk 3 above)
Comment (zmedico) [inline db/db.go]: Yes, those are removed now.

File: db/db.go [hunk 6]
```diff
@@ -77,65 +57,48 @@ type Rows struct {
 
 // Open an existing database, creating it if it does not exist.
 func Open(dbPath string) (*DB, error) {
-	openMu.Lock()
-	defer openMu.Unlock()
-
-	dbc, err := sql.Open("sqlite3_with_hook", dbPath)
+	err := os.MkdirAll(filepath.Dir(dbPath), 0755)
```
Comment (otoolep) [inline db/db.go]: Yeah, you can delete this now, since it's done at the `Store` code.

**Comment by otoolep** [comment]
Thanks again. Minor feedback. If I don't see changes soon, I'll merge yours and make the changes.

File: db/db.go (hunk 5 above)
Comment (zmedico) [inline db/db.go]: Done.

File: db/db.go (hunk 6 above)
Comment (zmedico) [inline db/db.go]: Done.

**Comment by otoolep** [comment]
Thanks @zmedico !

**Comment by zmedico** [comment]
I didn't push the changelog entry before you merged it.

**Comment by otoolep** [comment]
No worries @zmedico -- I updated the log for you.

**Pull request merged**